jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/DefaultServlet.java

```diff
@@ -861,18 +861,22 @@ private static class ServletCoreResponse implements Response
         private final ServletCoreRequest _coreRequest;
         private final Response _coreResponse;
         private final HttpFields.Mutable _httpFields;
+        private final boolean _included;
 
-        public ServletCoreResponse(ServletCoreRequest coreRequest, HttpServletResponse response)
+        public ServletCoreResponse(ServletCoreRequest coreRequest, HttpServletResponse response, boolean include)
         {
             _coreRequest = coreRequest;
             _response = response;
             _coreResponse = ServletContextResponse.getServletContextResponse(response);
             _httpFields = new HttpServletResponseHttpFields(response);
+            _included = include;
         }
 
         @Override
         public HttpFields.Mutable getHeaders()
         {
+            if (_included)
+                return HttpFields.build(_httpFields);
```

Comment:
Hmmm this is not exactly the right semantic.   It will not propagate sets to the real headers, but a set then a get would see the wrong updated value.   We need a special `HttpFields` wrapper that just ignores sets.

Reply:
You're right, let me fix that.